packages/ember/tests/routing/query_params_test/query_param_async_get_handler_test.js

```diff
@@ -0,0 +1,80 @@
+import {
+  Controller
+} from 'ember-runtime';
+
+import { QueryParamTestCase, moduleFor } from 'internal-test-helpers';
+
+moduleFor('Query Params - async get handler', class extends QueryParamTestCase {
+  get routerOptions() {
+    return {
+      location: 'test',
+      _getHandlerFunction() {
+        let getHandler = this._super(...arguments);
+        return (routeName) => Promise.resolve().then(() => getHandler(routeName));
```

Comment:
Likely should use off of `RSVP.Promise` (since we support platforms without native Promise).  Also, when you make that change you should use "real" async and make a helper method to resolve in a `setTimeout` or something...